virttest/utils_test/qemu/__init__.py

```diff
@@ -137,33 +137,30 @@ def _check_driver_verifier(session, driver, timeout=300):
 
 
 @error_context.context_aware
-def setup_win_driver_verifier(driver, vm, timeout=300):
+def setup_win_driver_verifier(session, driver, vm, timeout=300):
     """
     Enable driver verifier for windows guest.
 
     :param driver: The driver which needs enable the verifier.
     :param vm: VM object.
     :param timeout: Timeout in seconds.
     """
-    session = vm.wait_for_login(timeout=timeout)
-    try:
-        verifier_status = _check_driver_verifier(session, driver)[0]
+    verifier_status = _check_driver_verifier(session, driver)[0]
+    if not verifier_status:
+        error_context.context("Enable %s driver verifier" % driver,
+                              logging.info)
+        verifier_setup_cmd = "verifier /standard /driver %s.sys" % driver
+        session.cmd(verifier_setup_cmd,
+                    timeout=timeout,
+                    ignore_all_errors=True)
+        session = vm.reboot(session)
+        verifier_status, output = _check_driver_verifier(session, driver)
         if not verifier_status:
-            error_context.context("Enable %s driver verifier" % driver,
-                                  logging.info)
-            verifier_setup_cmd = "verifier /standard /driver %s.sys" % driver
-            session.cmd(verifier_setup_cmd,
-                        timeout=timeout,
-                        ignore_all_errors=True)
-            session = vm.reboot(session)
-            verifier_status, output = _check_driver_verifier(session, driver)
-            if not verifier_status:
-                msg = "%s verifier is not enabled, details: %s" % (driver,
-                                                                   output)
-                raise exceptions.TestFail(msg)
-        logging.info("%s verifier is enabled already" % driver)
-    finally:
-        session.close()
+            msg = "%s verifier is not enabled, details: %s" % (driver,
+                                                               output)
+            raise exceptions.TestFail(msg)
+    logging.info("%s verifier is enabled already" % driver)
+    return session
```

Comment:
hi, why return session?

Reply:
if guest reboot after driver verifier, should return session, since the session changed.